kerchunk/hdf.py

```diff
@@ -113,22 +113,39 @@ def __init__(
         self.error = error
         lggr.debug(f"HDF5 file URI: {self._uri}")
 
-    def translate(self):
+    def translate(self, preserve_linked_dsets=False):
         """Translate content of one HDF5 file into Zarr storage format.
 
         This method is the main entry point to execute the workflow, and
         returns a "reference" structure to be used with zarr/kerchunk
 
         No data is copied out of the HDF5 file.
 
+        Parameters
+        ----------
+        preserve_linked_dsets : bool (optional, default False)
+            If True, translate HDF5 soft and hard links for each `h5py.Dataset`
+            into the reference structure. Requires h5py version 3.11.0 or later.
+            Will not translate external links or links to `h5py.Group` objects.
+
         Returns
         -------
         dict
             Dictionary containing reference structure.
         """
         lggr.debug("Translation begins")
         self._transfer_attrs(self._h5f, self._zroot)
+
         self._h5f.visititems(self._translator)
+
+        if preserve_linked_dsets:
+            if h5py.__version__ < "3.11.0":
```

Comment:
Should we use more sophisticated version checking here? String comparisons are notoriously flaky.